dotCMS/src/main/java/com/dotcms/ai/api/EmbeddingsAPIImpl.java

```diff
@@ -33,191 +37,208 @@
 
 import javax.validation.constraints.NotNull;
 import java.time.Duration;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 import java.util.stream.Collectors;
 
-
 class EmbeddingsAPIImpl implements EmbeddingsAPI {
 
+    private static final Cache<String, Tuple2<Integer, List<Float>>> EMBEDDING_CACHE =
```

Comment:
any reason using the Caffeine implementation instead of the CacheLocator?
I think this is limiting the chain abilities and others functionalities

Reply:
2 reasons 
- I am not sure we need this and it could be considered a premature optimization.
- Because this was coming from a plugin where we did not have access to add new CacheRegions.

I like to use Caffeine whenever I need an ExpiringMap because I know it is hardened and well tested.